### PR TITLE
Fix remaining sphinx documentation

### PR DESCRIPTION
**Pull request by gassmoeller**
This PR builds on #5158 (so the other one should be merged first) and should finish the biggest part of #4595. 

With this PR the conversion to sphinx and markdown based documentation should be complete (there are improvements we can make, but the critical points are done). 
The PR in particular fixes the remaining warnings and removes the "under construction" signs on the website. 
Some problems that I fixed are:
- We had README.md files that were not included in the documentation, sphinx creates a warning about them. I integrated the files into the documentation structure (which also means we now have more of our benchmarks in the documentation)
- All sphinx files need to start at header level 1 (`#`)
- fixed a bunch of typos in section labels or added missing labels
- included way more direct links to directories or files where they are mentioned
- unified the citation style in a number of places
- improved the `jsontomarkdown.py` file to escape the rare character combinations that can create a problem for markdown (like `[abd](...) = ...` which looks like a link to markdown)
- removed all cases where heading levels were non-consecutive (i.e. a document would start with `# bla` and continue with `### bla`)
- integrated the `quickref.md` markdown reference into the structure of our documentation. For one we want to keep this file for the future, and secondly this fixes some broken links inside of that file
- Sphinx warnings are now treated as errors, so future PRs that create warnings will fail the documentation tester

### CODE REVIEW

File: cookbooks/composition-reaction/doc/composition-reaction.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## The active case with reactions.
+# The active case with reactions.
```
Comment (tjhei) [inline cookbooks/composition-reaction/doc/composition-reaction.md:1]: Why do we need to make this change? Logically they are subchapters to  https://aspect-documentation.readthedocs.io/en/latest/user/cookbooks/cookbooks-composition.html or not?

**Review by tjhei — APPROVED** [review]
Looks good, thank you for working on this!

File: cookbooks/composition-reaction/doc/composition-reaction.md (hunk 1 above)
Comment (gassmoeller) [reply cookbooks/composition-reaction/doc/composition-reaction.md:1]: They are, but Sphinx expects all documents to start with a `#` title. This means the `#` levels determine the rendering and logical ordering of headings on a particular page (or one markdown file), and the appearance of the file within the `toctree` sections determines where it is logically in the global table of contents.

**Comment by gassmoeller** [comment]
@tjhei is this ready to merge? I just rebased after #5160.

**Comment by gassmoeller** [comment]
Thanks!

**Pull request merged**